src/wazuh_modules/task_manager/wm_task_manager.c

```diff
@@ -252,6 +281,38 @@ cJSON* wm_task_manager_analyze_task(const cJSON *task_object, int *error_code) {
                 response = wm_task_manager_build_response(WM_TASK_INVALID_AGENT_ID, agent_id, task_id, status);
             }
 
+        } else if (!strcmp(commands_list[WM_TASK_UPGRADE_RESULT], command)) {
+            
+            if (agent_id != OS_INVALID) {
+                if (result = wm_task_manager_get_task_by_agent_id_and_module(agent_id, modules_list[WM_TASK_UPGRADE_MODULE], &command_result, &status, &create_time, &last_update_time), result == OS_INVALID){
+                    *error_code = WM_TASK_DATABASE_ERROR;
+                    response = wm_task_manager_build_response(WM_TASK_INVALID_AGENT_ID, agent_id, result, status);
+                } else if (result == OS_NOTFOUND || result == 0){
+                    *error_code = WM_TASK_DATABASE_NO_AGENT;
+                    response = wm_task_manager_build_response(WM_TASK_DATABASE_NO_AGENT, agent_id, OS_INVALID, status);
+                } else {
+                    response = wm_task_manager_build_response(WM_TASK_SUCCESS, agent_id, result, NULL);
+                    wm_task_manager_build_response_result(response, modules_list[WM_TASK_UPGRADE_MODULE], command_result, status, create_time, last_update_time, command);
+                    os_free(command_result);
+                }
+            } else {
+                *error_code = WM_TASK_INVALID_AGENT_ID;
+                response = wm_task_manager_build_response(WM_TASK_INVALID_AGENT_ID, agent_id, task_id, status);
+            }
+        } else if (!strcmp(commands_list[WM_TASK_TASK_RESULT], command)) {
+
+            if (result = wm_task_manager_get_task_by_task_id(task_id, &agent_id, &module_result, &command_result, &status, &create_time, &last_update_time), result == OS_INVALID){
```

Comment:
Idem here.

Reply:
here I need to keep the two variables separate because if there was an error or it didn't find the task, it would lose the value of task_id to send the response message